cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -321,6 +385,13 @@ predicate localFlowStep(Node nodeFrom, Node nodeTo) { simpleLocalFlowStep(nodeFr
  */
 predicate simpleLocalFlowStep(Node nodeFrom, Node nodeTo) {
   simpleInstructionLocalFlowStep(nodeFrom.asInstruction(), nodeTo.asInstruction())
+  or
+  exists(LoadInstruction load, ChiInstruction chi |
+    not chi.isResultConflated() and
+    nodeTo.asInstruction() = load and
+    nodeFrom.asInstruction() = chi and
+    load.getSourceValueOperand().getAnyDef() = chi
+  )
```

Comment:
You can move this case into `simpleInstructionLocalFlowStep`. That will save you two `asInstruction` calls, and it will make this code `cached` when #3189 is merged.

Reply:
I've done this now in d56284f. This case I've added in `simpleInstructionLocalFlowStep` has an obvious generalization, which we might add at some later stage. But I think that deserves its own investigation first.